packages/webamp/js/components/EqualizerWindow/index.tsx

```diff
@@ -35,6 +37,124 @@ const EqualizerWindow = () => {
     Actions.toggleEqualizerShadeMode
   );
 
+  /**
+   * getBandFromClientX()
+   * Return the band # (60,170,310,etc.) that we are over during the touchmove.
+   * We do this by taking the first band "band-60", and using it's left pos and width
+   *  and calculating the rest of the bands from there.
+   * In this routine we only care about the x axis.
+   * @param clientX 
+   * @returns Band
+   */
+  const getBandFromClientX = (clientX: number) => {
+    // This routine is very much dependent on the first eq band 60 and it's coordinates.
+    // We have to bail out if we can't get that information.
+    const band_60_rect = document.getElementById("band-60")?.getBoundingClientRect();
+    if (!band_60_rect) { return null; }
+
+    const startX = band_60_rect.left;
+    // It would be nice if stepX was simply the width of each VerticalSlider.
+    // However the numbers are a bit off, both in doubled and normal modes.
+    // So we add a magic number (sorry) in both doubled and normal mode to line things up better.
+    const magicNumber = doubled ? 7 : 4;
+    const stepX = band_60_rect.width + magicNumber;
+
+    // One nice thing about doing it this way, when you touch drag far left beyond the
+    // leftmost slider, it will lock to the leftmost. Similarly, when you touch drag far
+    // right of the rightmost slider, it will lock to the rightmost.
+    let x = clientX;
+    if (x < startX + (stepX * 1)) {
+      return 60;
+    } else if (x >= startX + (stepX * 1) && x < startX + (stepX * 2)) {
+      return 170;
+    } else if (x >= startX + (stepX * 2) && x < startX + (stepX * 3)) {
+      return 310;
+    } else if (x >= startX + (stepX * 3) && x < startX + (stepX * 4)) {
+      return 600;
+    } else if (x >= startX + (stepX * 4) && x < startX + (stepX * 5)) {
+      return 1000;
+    } else if (x >= startX + (stepX * 5) && x < startX + (stepX * 6)) {
+      return 3000;
+    } else if (x >= startX + (stepX * 6) && x < startX + (stepX * 7)) {
+      return 6000;
+    } else if (x >= startX + (stepX * 7) && x < startX + (stepX * 8)) {
+      return 12000;
+    } else if (x >= startX + (stepX * 8) && x < startX + (stepX * 9)) {
+      return 14000;
+    } else {
+      return 16000;
+    }
+  };
+
+  /**
+   * setWinampActiveClass()
+   * Set ACTIVE_CLASSNAME on the current band (to set the selected button state)
+   */
+  const setWinampActiveClass = (band: BandType) => {
+    document.querySelector(`#band-${band}`)?.classList.add(ACTIVE_CLASSNAME);
```

Comment:
Let's try to find a solution that uses React rendering to set the DOM state rather than manually manipulating the DOM. If manual DOM manipulation is absolutely required (I don't think it will be), it should happen within the component that rendered the DOM element.

Reply:
This is fixed in local dev and this hack is no longer necessary